### PR TITLE
Install GitHub

### PR DESCRIPTION
**Pull request by JanCaha**
This pull request introduces a new metadata line that allows the installation of R packages from GitHub via R package `remotes`.

The format is:

`
##JanCaha/CzechData, JanCaha/SpatialKDE=github_install
`

Which is processed just before the packages are loaded.

This quite likely very useful as there is a significant amount of packages that are not available from CRAN (for various reasons). This allows users to use these packages and declare this use formally.

### CODE REVIEW

File: processing_r/processing/algorithm.py [hunk 1]
```diff
@@ -456,13 +465,22 @@ def build_script_header_commands(self, _, __, ___):
...
+        if self.r_templates.install_github:
+            for dependency in self.r_templates.github_dependencies:
+                commands.append(self.r_templates.install_package_github(dependency))
+
+        packages_script = RUtils.get_required_packages(self.script)
+
+        for p in packages_script:
+            commands.append(self.r_templates.check_package_availability(p))
+            commands.append(self.r_templates.load_package(p))
+
```
Comment (nyalldawson) [inline processing_r/processing/algorithm.py]: I think this whole block of code could be moved to r_templates -- what do you think?

**Comment by nyalldawson** [comment]
@JanCaha we should add this to the documentation too. When it's merged I'll push out a new release

File: processing_r/processing/algorithm.py (hunk 1 above)
Comment (JanCaha) [reply processing_r/processing/algorithm.py]: That makes sense, all the info is contained there anyway! I will fix it later today :)

Btw: Just out of curiosity, is there a way to add the changes to this pull request or do I have to close it and make a new one?

Comment (nyalldawson) [reply processing_r/processing/algorithm.py]: If you commit them to the same branch and then push, they'll appear here automatically!

**Comment by nyalldawson** [comment]
Thanks! Happy for me to push a new release off master?

**Comment by JanCaha** [comment]
Of course :)

Hopefully, I will remember until the morning to update the Changelog for the website ...

**Pull request merged**